conformance/lang/expressions/regexp/regexp_semantics.balt

```diff
@@ -0,0 +1,719 @@
+    io:println(reg5 is regexp:RegExp); // @output true
+
+    any & readonly reg6 = reg1;
+    io:println(reg6 is string:RegExp); // @output true
+    io:println(reg6 is regexp:RegExp); // @output true
+}
+
+Test-Case: output
+Description: Test regular expression with user defined types.
+Labels: RegExp, module-type-defn
+
+type UserDefType string:RegExp; 
+
+function init() {
+    UserDefType reg1 = re `([A-D]+|[H-Z]*)|C{2,9}`;
+    io:println(reg1 is string:RegExp); // @output true
+    io:println(re `([A-D]+|[H-Z]*)|C{2,9}` == reg1); // @output true
+}
+
+Test-Case: output
+Description: Test regular expression with user defined union types.
+Labels: RegExp, module-type-defn
+
+type UserDefUnionType string:RegExp|json|map<anydata>; 
+
+function init() {
+    UserDefUnionType reg1 = re `([A-D]+|[H-Z]*)|C{2,9}`;
+    io:println(reg1 is string:RegExp); // @output true
+    io:println(re `([A-D]+|[H-Z]*)|C{2,9}` == reg1); // @output true
+}
+
+Test-Case: output
+Description: Test subtyping with regular expression.
+Labels: RegExp, module-type-defn, subtyping
+
+type UserDefUnionType string:RegExp|string|map<anydata>; 
+
+function init() {
+    string:RegExp reg1 = re `([A-D]+|[H-Z]*)|C{2,9}`;
+    io:println(reg1 is anydata); // @output true
+    io:println(reg1 is any); // @output true
+    io:println(reg1 is anydata); // @output true
+    
+    anydata reg2 = re `([A-D]+|[H-Z]*)|C{2,9}`;
+    io:println(reg2 is string:RegExp); // @output true
+
+    any reg3 = re `([A-D]+|[H-Z]*)|C{2,9}`;
+    io:println(reg3 is string:RegExp); // @output true
+
+    readonly reg4 = re `([A-D]+|[H-Z]*)|C{2,9}`;
+    io:println(reg4 is string:RegExp); // @output true
+
+    anydata & readonly reg5 = re `([A-D]+|[H-Z]*)|C{2,9}`;
+    io:println(reg5 is string:RegExp); // @output true
+
+    any & readonly reg6 = re `([A-D]+|[H-Z]*)|C{2,9}`;
+    io:println(reg6 is string:RegExp); // @output true
+
+    anydata & readonly reg7 = re `([A-D]+|[H-Z]*)|C{2,9}`;
+    io:println(reg7 is string:RegExp); // @output true
+
+    UserDefUnionType reg8 = re `([A-D]+|[H-Z]*)|C{2,9}`;
+    io:println(reg8 is string:RegExp); // @output true
+}
+
+Test-Case: output
+Description: Test regular expression as function parameter.
+Labels: RegExp, function-type-descriptor
+
+function getRegExp(string:RegExp re) returns string:RegExp {
+    return re;
+}
+
+function init() {
+    string:RegExp reg1 = re `([A-D]+|[H-Z]*)|C{2,9}`;
+    io:println(getRegExp(reg1) == re `([A-D]+|[H-Z]*)|C{2,9}`); // @output true
+
+    string:RegExp reg2 = getRegExp(re `([A-D]+|[H-Z]*)|C{2,9}`);
+    io:println(reg2 == re `([A-D]+|[H-Z]*)|C{2,9}`); // @output true
+
+    string:RegExp reg3 = getRegExp(re = re `([A-D]+|[H-Z]*)|C{2,9}`);
+    io:println(reg3 == re `([A-D]+|[H-Z]*)|C{2,9}`); // @output true
+}
+
+Test-Case: output
+Description: Test regular expression as function defaultable parameter.
+Labels: RegExp, defaultable-params, function-type-descriptor
```

Comment:
Should be `defaultable-param`.
We have not used a `function-type-descriptor` here?